# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1425404321 18000
#      Tue Mar 03 12:38:41 2015 -0500
# Node ID ac18e88ea1bf5efdeb591c4d917d32d13419c8e0
# Parent  67cb2f42125569879434acb33c64f27194286ec6
zoom about point for single clicks

* Canvas.cc (Canvas::canvasMouseReleaseEvent): Zoom about point for
single click zooming.

diff --git a/libgui/graphics/Canvas.cc b/libgui/graphics/Canvas.cc
--- a/libgui/graphics/Canvas.cc
+++ b/libgui/graphics/Canvas.cc
@@ -663,17 +663,19 @@ void Canvas::canvasMouseReleaseEvent (QM
           graphics_object figObj (obj.get_ancestor ("figure"));
 
           std::string zm = zoom_mode (figObj);
 
           if (m_mouseAnchor == event->pos ())
             {
               double factor = m_mouseMode == ZoomInMode ? 2.0 : 0.5;
 
-              ap.zoom (zm, factor);
+              ColumnVector p1 = ap.pixel2coord (event->x (), event->y ());
+
+              ap.zoom_about_point (zm, p1(0), p1(1), factor);
             }
           else
             {
               ColumnVector p0 = ap.pixel2coord (m_mouseAnchor.x (),
                                                 m_mouseAnchor.y ());
               ColumnVector p1 = ap.pixel2coord (event->x (),
                                                 event->y ());
 
